README.md

```diff
@@ -640,6 +640,42 @@ note: Step 3: call itself without input consumption - left recursion
   |        ^^^^^
 ```
 
+## Plugins API
+
+Plugin is an object with the `use(config, options)` method. That method will be
+called for all plugins in the `options.plugins` array, supplied to the `generate()`
+method.
+
+Plugin accepts the following parameters:
+- `config` — object with the following properties:
+  - `parser` — `Parser` object, by default the `peg.parser` instance. That object
+    will be used to parse the grammar. Plugin can replace this object
+  - `passes` — mapping `{ [stage: string]: Pass[] }` that represents compilation
```

Comment:
this points out that [compiler.compile](https://github.com/peggyjs/peggy/blob/42a133729393f91b4a74aa587417857600303e62/lib/compiler/index.js#L75) and the code that uses it in [peg.js](https://github.com/peggyjs/peggy/blob/42a133729393f91b4a74aa587417857600303e62/lib/peg.js#L30) is overly-confusing and relies on `Object.keys` to be consistent in the face of plugins making changes to the passes.

Let's look at that code as we document this interface, and see if it needs work.

Reply:
Actually, according to the MDN, all works because of implicit assumption that keys always will be returned in the insertion order (which is equal to the declaration order for object literals). But I thin that this is implementation detail, conceptually my description should remains unchanged even if we change implementation.

Reply:
As a plugin author, I want to add a pass that is between `check` and `transform`.  I guess I could do it without breaking other plugins that did the same thing by creating a new object, traversing `Object.keys(passes)`, inserting my pass after `check`,  and re-assigning passes.  I guess that's close enough.